HopsanGUI/Widgets/LibraryWidget.cpp

```diff
@@ -461,7 +463,12 @@ void LibraryWidget::handleItemClick(QTreeWidgetItem *item, int column)
         {
             pRecompileAction->setVisible(true);
             pEditXMLAction->setVisible(true);
-            pEditCodeAction->setVisible(true);
+            if(!gpLibraryHandler->getEntry(mItemToTypeNameMap.find(item).value()).pAppearance->getSourceCodeFile().isEmpty()) {
```

Comment:
since this long line of code is reused below, cosider saving the result in a bool.
"itemHasSourceCode" and use that bool in the setVisible call

Same for  hmfFile, even though its only used once.

But I could not test this menu item, it never showed up for me. I took the subsystem example model and saved Servo1 as a component inside the defaultLibrary/Special
I can not get this menu option, not sure why, have not had time to debug